MetadataExtractor/Formats/Jpeg/HuffmanTablesDirectory.cs

```diff
@@ -0,0 +1,323 @@
+        }
+
+        /// <returns>The number of Huffman tables held by this HuffmanTablesDirectory instance.</returns>
+        public int GetNumberOfTables()
+        {
+            return this.GetInt32(TagNumberOfTables);
+        }
+
+        /// <returns>The List of HuffmanTables in this Directory.</returns>
+        [NotNull]
+        public List<HuffmanTable> Tables { get; set; } = new List<HuffmanTable>(4);
+
+        /// <summary>Evaluates whether all the tables in this HuffmanTablesDirectory are "typical" Huffman tables.</summary>
+        /// <remarks>
+        /// "Typical" has a special meaning in this context as the JPEG standard
+        /// (ISO/IEC 10918 or ITU-T T.81) defines 4 Huffman tables that has been
+        /// developed from the average statistics of a large set of images with 8-bit
+        /// precision. Using these instead of calculating the optimal Huffman tables
+        /// for a given image is faster, and is preferred by many hardware encoders
+        /// and some hardware decoders.
+        /// <para/>
+        /// Even though the JPEG standard doesn't define these as "standard tables"
+        /// and requires a decoder to be able to read any valid Huffman tables, some
+        /// are in reality limited decoding images using these "typical" tables.
+        /// Standards like DCF(Design rule for Camera File system) and DLNA(Digital
+        /// Living Network Alliance) actually requires any compliant JPEG to use only
+        /// the "typical" Huffman tables.
+        /// <para/>
+        /// This is also related to the term "optimized" JPEG. An "optimized" JPEG is
+        /// a JPEG that doesn't use the "typical" Huffman tables.
+        /// </remarks>
+        public bool IsTypical()
+        {
+            if (Tables.Count == 0)
+            {
+                return false;
+            }
+            foreach (HuffmanTable table in Tables)
+            {
+                if (!table.IsTypical())
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        /// <remarks>The opposite of IsTypical().</remarks>
+        /// <returns>
+        /// Whether or not the tables in this HuffmanTablesDirectory
+        /// are "optimized" - which means that at least one of them aren't
+        /// one of the "typical" Huffman tables.
+        /// </returns>
+        public bool IsOptimized()
+        {
+            return !IsTypical();
+        }
+
+    }
+
+    /// <summary>An instance of this class holds a JPEG Huffman table.</summary>
+    public class HuffmanTable
+    {
+        private byte[] _lengthBytes;
+        private byte[] _valueBytes;
+
+        public HuffmanTable([NotNull] HuffmanTableClass tableClass, int tableDestinationId, [NotNull] byte[] lengthBytes, [NotNull] byte[] valueBytes)
+        {
+            _lengthBytes = lengthBytes ?? throw new ArgumentNullException("lengthBytes cannot be null.");
```

Comment:
The argument here (and the line below) are supposed to be the argument name. The exception formulates an appropriate string in the user's language.

```suggestion
            _lengthBytes = lengthBytes ?? throw new ArgumentNullException(nameof(lengthBytes));
```